### PR TITLE
- added opa bundle builder (makefile) and opa OPA_BUILDER_CLUSTERROLE…

### PR DESCRIPTION
**Pull request by maltesander**
… (deployment)

- removed obsolete extra_crates
- removed spaces from readme

### CODE REVIEW

File: template/Makefile.j2 [hunk 1]
```diff
@@ -15,6 +15,9 @@ VERSION := $(shell cargo metadata --format-version 1 | jq '.packages[] | select(
 ## Docker related targets
 docker-build:
 	docker build --force-rm -t "docker.stackable.tech/stackable/{[ operator.product_string }]-operator:${VERSION}" -f docker/Dockerfile .
+{[% if operator.product_string in ['opa'] %}]
```
Comment (soenkeliebau) [inline template/Makefile.j2:18]: This is probably ok for now, but if we start adding more of these conditions here I think I would prefer adding a "dockerfiles" variable that defaults to `["docker/Dockerfile"]` but can be overridden to allow building and pushing multiple docker images here.

File: template/deploy/helm/[[product]]-operator/templates/deployment.yaml.j2 [hunk 2]
```diff
@@ -44,6 +44,11 @@ spec:
           env:
             - name: KAFKA_BROKER_CLUSTERROLE
               value: {{ .Release.Name }}-kafka-broker-clusterrole
+{[% endif %}]
+{[% if operator.product_string in ['opa'] %}]
+          env:
+            - name: OPA_BUILDER_CLUSTERROLE
+              value: {{ .Release.Name }}-opa-builder-clusterrole
```
Comment (soenkeliebau) [inline template/deploy/helm/[[product]]-operator/templates/deployment.yaml.j2:51]: nit: should this not be called bundle-builder ?

**Review by soenkeliebau — APPROVED** [review]
two comments, but overall looks good to me

File: template/Makefile.j2 [hunk 3]
```diff
@@ -15,6 +15,9 @@ VERSION := $(shell cargo metadata --format-version 1 | jq '.packages[] | select(
...
+	docker build --force-rm -t "docker.stackable.tech/stackable/{[ operator.product_string }]-bundle-builder:${VERSION}" -f docker/Dockerfile.builder .
+{[% endif %}]
 
 docker-build-latest: docker-build
 	docker tag "docker.stackable.tech/stackable/{[ operator.product_string }]-operator:${VERSION}" \
```
Comment (soenkeliebau) [inline template/Makefile.j2:23]: Does this also need changing for the bundle builder?

**Review by soenkeliebau — CHANGES_REQUESTED** [review]
Actually have one question that I'd like to clarify before merging..

File: template/Makefile.j2 (hunk 3 above)
Comment (razvan) [reply template/Makefile.j2:23]: We can it it's not needed. The image for the builder is never installed from `latest` but always from the exact same version as the operator.
To put it differently: the builder is image is a "product image" that always comes with the same version as the operator.

File: template/deploy/helm/[[product]]-operator/templates/deployment.yaml.j2 (hunk 2 above)
Comment (razvan) [reply template/deploy/helm/[[product]]-operator/templates/deployment.yaml.j2:51]: Yeah it should.

File: template/Makefile.j2 (hunk 3 above)
Comment (soenkeliebau) [reply template/Makefile.j2:23]: ok

**Review by soenkeliebau — DISMISSED** [review]

File: template/deploy/helm/[[product]]-operator/templates/deployment.yaml.j2 (hunk 2 above)
Comment (razvan) [reply template/deploy/helm/[[product]]-operator/templates/deployment.yaml.j2:51]: Done.

**Review by razvan — APPROVED** [review]
lgtm

**Pull request merged**